pkg/autodiscovery/procdiscovery/process_nowindows.go

```diff
@@ -0,0 +1,29 @@
+// +build !windows
+
+package procdiscovery
+
+import (
+	"fmt"
+
+	goproc "github.com/shirou/gopsutil/process"
+)
+
+func pollProcesses() ([]process, error) {
+	processes, err := goproc.Processes()
+
+	if err != nil {
+		return nil, fmt.Errorf("Couldn't retrieve process list: %s", err)
+	}
+
+	pl := []process{}
```

Comment:
good to initialize slices with a capacity if you know it already to avoid extra allocs